common/models/entities_with_tags.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+type EntitiesWithTags []*EntityWithTags
+
+type ewtFilter func(e EntityWithTags) bool
+
+func (e EntitiesWithTags) RemoveIf(f ewtFilter) EntitiesWithTags {
```

Comment:
>  if someone external was to consume this package individually, would the logic contained here be helpful?

Yes, definitely. It's similar to adding python-like list semantics to a go slice, e.g. you can do things like `tags.First()` .

I get why being rigid on the rule that `/models` should only ever contain just structs has its merits. And if we should hardline on that, then ok let's move these to common 

But to put in my final thoughts -  I had the same feelings initially that I should put the logic stuff in `/common`. However, when you try to use it practice, it gets really annoying needing to to do this back-and-forth conversion between the logic and model packages. Functions end up looking like:
```
func getNameTag(type, id string) (*models.Tag, error) {
    tags, err := tagStore.Select(type, id)
    if err != nil { ... }

    // convert []models.Tag to common.Tags
    tagLogic := common.Tags(tags)

     // find by name
    nameTag := tagLogic.WithKey("name").First()

    // convert common.Tag back to models.Tag
    return models.Tag(nameTag), nil
}
```

vs. 

```
func getNameTag(type, id string) (*models.Tag, error) {
    tags, err := tagStore.Select(type, id)
    if err != nil { ... }

    return tags.WithKey("name").First(), nil
}
```

Also, I feel like our apprehension for putting any sort of logic into `/models` may be a result of us using the old `layer0-models` repository with swaggergen.